.github/workflows/devtools.yml

```diff
@@ -32,37 +37,74 @@ jobs:
     # with the latest dependencies. Any commit to the DevTools repo touching Lighthouse
     # code will invalidate the cache sooner.
     - name: Cache depot tools, devtools, blink tools and content shell
+      id: devtools-cache
       uses: actions/cache@v2
       with:
         path: |
           ${{ env.DEPOT_TOOLS_PATH }}
           ${{ env.DEVTOOLS_PATH }}
           ${{ env.BLINK_TOOLS_PATH }}
           ${{ github.workspace }}/lighthouse/.tmp/chromium-web-tests/content-shells
+          ${{ github.workspace }}/.gclient
         # This hash key changes:
         # 1) every monday (so invalidates once a week)
         # 2) every commit to CDT touching files important to Lighthouse web tests
         # 3) every change to file in Lighthouse repo important to running these tests.
         #
         # The number is how many times this hash key was manually updated to break the cache.
         key: ${{ runner.os }}-2-${{ env.WEEK_OF_THE_YEAR }}-${{ hashFiles('cdt-test-hash.txt') }}
-
-    - name: Use Node.js 14.x
-      uses: actions/setup-node@v1
-      with:
-        node-version: 14.x
+        restore-keys: ${{ runner.os }}-
+    - name: Set GHA_DEVTOOLS_CACHE_HIT
+      if: steps.devtools-cache.outputs.cache-hit == 'true'
+      run: echo "GHA_DEVTOOLS_CACHE_HIT=1" >> $GITHUB_ENV
 
     - run: yarn --frozen-lockfile
       working-directory: ${{ github.workspace }}/lighthouse
-    - run: yarn build-report
-      working-directory: ${{ github.workspace }}/lighthouse
-    - run: yarn build-devtools
-      working-directory: ${{ github.workspace }}/lighthouse
 
     - name: Download depot tools
       run: bash $GITHUB_WORKSPACE/lighthouse/lighthouse-core/test/chromium-web-tests/download-depot-tools.sh
     - name: Download DevTools Frontend
       run: bash $GITHUB_WORKSPACE/lighthouse/lighthouse-core/test/chromium-web-tests/download-devtools.sh
+
+    - name: Roll Lighthouse + build DevTools
+      run: bash $GITHUB_WORKSPACE/lighthouse/lighthouse-core/test/chromium-web-tests/roll-devtools.sh
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: devtools-build
+        path: |
+          ${{ env.DEVTOOLS_PATH }}/out/Default/gen/front_end
+          ${{ env.DEVTOOLS_PATH }}/test/webtests
+          ${{ github.workspace }}/.gclient
+        retention-days: 1
+
+  web-tests:
+    needs: [build]
+    runs-on: macos-latest
+
+    steps:
+    - name: Set $DEVTOOLS_PATH
+      run: echo "DEVTOOLS_PATH=$GITHUB_WORKSPACE/devtools-frontend" >> $GITHUB_ENV
+    - name: Set $BLINK_TOOLS_PATH
+      run: echo "BLINK_TOOLS_PATH=$GITHUB_WORKSPACE/blink_tools" >> $GITHUB_ENV
```

Comment:
looks like these can be set at the top level in an `env` block
```yml
env:
  DEPOT_TOOLS_PATH: $GITHUB_WORKSPACE/depot-tools
  DEVTOOLS_PATH: $GITHUB_WORKSPACE/devtools-frontend
  BLINK_TOOLS_PATH: $GITHUB_WORKSPACE/blink_tools
```
and so won't have to be repeated for each job